arviz/plots/bf_plot.py

```diff
@@ -0,0 +1,48 @@
+# Plotting and reporting Bayes Factor given idata, var name, prior distribution and reference value
+def plot_bf(trace, var_name, prior, family = 'normal',  ref_val=0, xlim=None, ax=None):
+    # grab trace, a variable name to compute difference and prior.
+    # ref_val is the parameter we want to compare
+    from scipy import stats
+    import matplotlib.pyplot as plt
+    import numpy as np
+    # test some elemtns
+    # varName should be string
+    if not isinstance(var_name, str):
+        print('varName is not a string')
+    # BFs based on density estimation (using kernel smoothing instead of spline)
+    # stack trace posterior
+    tr = trace.posterior.stack(draws=("chain", "draw"))
+    post = tr[var_name]
+    if post.ndim > 1:
+        print("Posterior distribution has {post.ndim} dimensions")
+    if family=='normal':
+        # generate vector
+        if xlim is None:
+            x = np.linspace(np.min(prior), np.max(prior),prior.shape[0])
+        else:
+            x = np.linspace(xlim[0], xlim[1],prior.shape[0])
+        #x = np.linspace(np.min(post), np.max(post),prior.shape[0])
+        my_pdf = stats.gaussian_kde(post)
+        prior_pdf = stats.gaussian_kde(prior)
+    elif family!='normal':
+        # for now and error of notImplemented
+        print("The method for {family} distribution is not implemented yet")
+    if ax is None:
+        fig, ax = plt.subplots()
+    ax.plot(
+        x, my_pdf(x), "--", lw=2.5, alpha=0.6, label="Posterior"
+    )  # distribution function
+    ax.plot(x, prior_pdf(x), "r-", lw=2.5, alpha=0.6, label="Prior")
```

Comment:
Is it necessary to use the same `x` when plotting the prior and the posterior? Most of the plotted line for the posterior will generally be the tail of the kde approximation. Does this method/figure (sorry for the lack of background again) work for bounded variables?

Reply:
Hi, Sorry for this very late response (I just started a position, so things are a bit hectic).
I think its better to use the same x, as we basically want to test the density under the same point of both prior and posterior.

Reply:
I think what Oriol is saying here is that we can build the kde using one x for the prior and one for the posterior to ensure we are placing points where there is some mass. Then we can evaluate those pdf for the same `ref_val`